### ISSUE TITLE
Reassign Parent for Keynote

### ISSUE THREAD

**Issue by mawdesign** [eirannejad/pyRevit#1920]
Would like to be able to change the parent of a keynote, we have a structure that is nested 3 levels so simply changing the category is not enough most of the time.

I have made it work under the "edit selected keynote" button by adding the following at around line 95 in Keynotes.pushbutton\script.py:
```
            #allow changing Parent key
            self.recordParent.IsEnabled = True
```
And then changing the pick_parent function at around line 266 to this:
```
    def pick_parent(self, sender, args):
        categories = kdb.get_categories(self._conn)
        keynotes = kdb.get_keynotes(self._conn)
        available_parents = [x.key for x in categories]
        available_parents.extend([x.key for x in keynotes])
        # remove self from that record if self is not none
        if self.active_key in available_parents:
            available_parents.remove(self.active_key)
        # prompt to select a record
        new_parent = forms.SelectFromList.show(
            available_parents,
            title='Select Parent',
            multiselect=False
            )
        if new_parent:
            try:
                kdb.reserve_key(self._conn, self.active_key, category=self._cat)
            except System.TimeoutException as toutex:
                forms.alert(toutex.Message)
                return
            self._reserved_key = self.active_key
            # apply the record key on the button
            self.active_parent_key = new_parent
```

I'm not keen to figure out pull requests yet and would prefer someone with a bit more experience with the pyRevit code incorporated it.

Thanks, Michael.

**Comment by jmcouffin** [comment]
@mawdesign 
Thanks, 
Please try it and report if behaviour is as expected
you can get the installers from the bottom of this page https://github.com/eirannejad/pyRevit/actions/runs/6196250744 in about 15min

**Comment by mawdesign** [comment]
Awesome. I should say that this only enables the Parent key button on the Edit selected keynote dialog, I'm not sure if that is the right, or only, place for it but it made sense to me. I tested it on a larger keynote file and realised that the order in the select list is just as they are in the keynote file. Adding natsort() makes for a more coherant list. I added
```
from natsort import natsorted
```
at the top of the file, and changed the select list function to:
```
        # prompt to select a record
        new_parent = forms.SelectFromList.show(
            natsorted(available_parents),
            title='Select Parent',
            multiselect=False
            )
```
Otherwise it worked perfectly.

**Comment by jmcouffin** [comment]
fixed by #1927
